Sql Assignment Day 3-C.sql

```diff
@@ -0,0 +1,78 @@
+-- Library Management Schema
+
+CREATE TABLE Authors (
+    AuthorID INT PRIMARY KEY,
+    AuthorName VARCHAR(100),
+    BirthYear INT
+);
+
+CREATE TABLE Books (
+    BookID INT PRIMARY KEY,
+    Title VARCHAR(200),
+    AuthorID INT,
+    PublicationYear INT,
+    FOREIGN KEY (AuthorID) REFERENCES Authors(AuthorID)
+);
+
+CREATE TABLE Patrons (
+    PatronID INT PRIMARY KEY,
+    PatronName VARCHAR(100),
+    MembershipDate DATE
+);
+
+CREATE TABLE Loans (
+    LoanID INT PRIMARY KEY,
+    BookID INT,
+    PatronID INT,
+    LoanDate DATE,
+    ReturnDate DATE,
+    FOREIGN KEY (BookID) REFERENCES Books(BookID),
+    FOREIGN KEY (PatronID) REFERENCES Patrons(PatronID)
+);
+
+-- Questions
+
+-- 1. List all books along with their authors, including books without assigned authors.
+SELECT BookID, Title, a.AuthorID, a.AuthorName
+FROM Books b LEFT JOIN Authors a ON b.AuthorID = a.AuthorID;
+
+-- 2. Display all patrons and their loan history, including patrons who have never borrowed a book.
+SELECT p.PatronID, PatronName, LoanID, BookID, LoanDate,ReturnDate
+FROM Patrons p LEFT JOIN Loans l ON p.PatronID = l.PatronID ;
+
+-- 3. Show all authors and the books they've written, including authors who haven't written any books in our collection.
+SELECT a.AuthorID, AuthorName, BookID, Title
+FROM Authors a LEFT JOIN Books b ON a.AuthorID = b.AuthorID;
+
+-- 4. List all possible book-patron combinations, regardless of whether a loan has occurred.
+SELECT BookID, Title, PatronID, PatronName
+FROM Books CROSS JOIN Patrons;
+
+-- 5. Display all loans with book and patron information, including loans where either the book or patron information is missing.
+SELECT  LoanID, b.BookID, Title, AuthorID, PublicationYear, p.PatronID, PatronName
+FROM Loans l LEFT JOIN Books b ON l.BookID = b.BookID
+LEFT JOIN Patrons p ON l.PatronID = p.PatronID;
+
+
+-- 6. Show all books that have never been loaned, along with their author information.
+SELECT b.BookID, Title, a.AuthorID, AuthorName, BirthYear
+FROM Loans l RIGHT JOIN Books b ON l.BookID = b.BookID 
+JOIN Authors a ON a.AuthorID = b.AuthorID
```

Comment:
Start from books

Reply:
corrected to :
...Books b LEFT JOIN Loans l...